doc/sphinx/proof-engine/ltac2.rst

```diff
@@ -154,7 +154,7 @@ One can define new types by the following commands.
    Records are product types with named fields and eliminated by projection.
    Likewise they can be recursive if the `rec` flag is set.
 
-   .. cmdv:: Ltac2 Type @ltac2_typeparams @ltac2_qualid := [ @ltac2_constructordef ]
+   .. cmdv:: Ltac2 Type {? @ltac2_typeparams } @ltac2_qualid := [ @ltac2_constructordef ]
```

Comment:
@ppedrot This is still incorrect: the command expects an unqualified identifier. However, this looks more like a Ltac2 bug than a documentation bug to me. Indeed, how is one supposed to expand, e.g., the `exn` type if it can only be an unqualified identifier?

```coq
From Ltac2 Require Import Ltac2.

Ltac2 Type exn := [ Foo ].

Ltac2 Eval Control.zero Foo.
```

gives the following error message:

```
This expression has type exn but an expression was expected of type Init.exn
```

Reply:
The syntax used for extension is not the same, it needs to use `::=` instead of `:=`. What you have done here is defining a new type with a brand new constructor, hence the error.

Reply:
Then the doc was incorrect since it was showing a syntax with only one colon. Fixed now.